jalib/jalloc.cpp

```diff
@@ -36,6 +38,7 @@ static bool _initialized = false;
 JAllocArena allocArenas[MAX_ARENAS];
 static int numAllocArenas;
 
+#undef JALIB_ALLOCATOR
```

Comment:
This is undefining JALIB_ALLOCATOR for _all_ CPUs, which is bad.

Also, please test this again.  I think you'll find that the 'main' branch now handles JALIB fine, in a way that's independent of the CPU.